src/graphics/canvas.rs

```diff
@@ -563,29 +563,22 @@ struct DrawCommand {
 #[derive(Debug)]
 pub(crate) struct DefaultResources {
     pub shader: Shader,
-    pub text_shader: Shader,
     pub mesh: Mesh,
     pub image: Image,
 }
 
 impl DefaultResources {
     fn new(gfx: &GraphicsContext) -> Self {
         let shader = Shader {
-            fragment: gfx.draw_shader.clone(),
-            fs_entry: "fs_main".into(),
-        };
-
-        let text_shader = Shader {
-            fragment: gfx.text_shader.clone(),
-            fs_entry: "fs_main".into(),
+            fs_module: None,
+            vs_module: None,
         };
 
         let mesh = gfx.rect_mesh.clone();
         let image = gfx.white_image.clone();
 
         DefaultResources {
             shader,
-            text_shader,
```

Comment:
Yeah... I was thinking of deleting it and using a `Shader {}` directly everwhere but I really don't know what would be clearest here.

Reply:
I'm not sure about the best way here either, but I'd say we delete the shader from this defaults struct.

I think it would be best to specify such defaults (sampler is set as linear clamp explicitly everywhere as well, so it's a similar case) in pub static functions `default_sampler` and `default_shader`, `default_text_shader`, so we still have the defaults of everything in one place instead of duplicating them.